### PR TITLE
docs(access policies): Creating Proper Access Policies Guide

### PR DESCRIPTION
**Pull request by jjoyce0510**
**Summary**
In this PR, we create an Access Policies guide that is more digestible than what we currently have, including step-by-step instructions on creating policies. 

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: docs/authorization/access-policies-guide.md [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+In the first step, we'll give our Policy a name and description. We'll also select the *type* of the policy,
+which determines which privileges can be granted using it.
+
+First give the new Policy a name.
+
+<p align="center">
+  <img width="40%"  src="https://raw.githubusercontent.com/datahub-project/datahub/master/docs/imgs/create-policy.png"/>
+</p>
+
+Next, select a *type* for the Policy. There are 2 types of Access Policy within DataHub:
```
Comment (maggiehays) [inline docs/authorization/access-policies-guide.md:53]: This is confusing:

> Step 1. Choose Policy Type
> In the first step, we'll give our Policy a name and description

Consider splitting out Name & Description as Step 1 & providing some guidance around the naming conventions. What are common naming patterns we recommend?

Then, move on to Step 2: Choose Policy Type

File: docs/authorization/access-policies-guide.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+1. **Privileges**: The 'what'. What actions are being permitted by a policy, e.g. "Add Tags".
+2. **Resources**: The 'which'. Resources that the policy applies to, e.g. "All Datasets".
+3. **Actors**: The 'who'. Specific users, groups, & roles that the policy applies to.
+
+Depending on the type of privileges you want to grant, select either
+the **Platform** or **Metadata** policy type.
+
+Finally, provide a description for the Policy to help keep track of its purpose.
+
+When you're done, click **Next** to continue to the next step.
```
Comment (maggiehays) [inline docs/authorization/access-policies-guide.md:92]: Do we need this level of instruction?

File: docs/authorization/access-policies-guide.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+| Create Tags                     | Allow the actor to create new Tags                                                                                             |
+| Manage Tags                     | Allow the actor to create and remove any Tags                                                                                  |
+
+*Metadata Policies*
+
+If we've chosen a Metadata Policy, we will first determine which assets that the privileges should be granted for (i.e. the *scope*), then
+we'll select privileges to grant.
+
+First, we can narrow down the *type* of the assets that the policy applies to. If left, blank
+all entity types will be in scope.
```
Comment (maggiehays) [inline docs/authorization/access-policies-guide.md:133]: > If left, blank

Update to:

> If left blank,

File: docs/authorization/access-policies-guide.md [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+This platform privilege allows users to create, edit, and remove all Access Policies on DataHub. Therefore, it should only be
+given to those users who will be serving as Admins of the platform. The default `Admin` role has this privilege.
+
+
+## Using Access Policies
+
+### Creating a new Policy
+
+Policies can be created by first navigating to **Settings > Permissions > Policies**. The `Policies` tab will only
+be visible to those users having the required `Manage Policies` privilege.
```
Comment (maggiehays) [inline docs/authorization/access-policies-guide.md:34]: Consider removing - this is already covered in the prerequisites section, above 

> The `Policies` tab will only be visible to those users having the required `Manage Policies` privilege.

File: docs/authorization/access-policies-guide.md [hunk 5]
```diff
@@ -0,0 +1,312 @@
+# About DataHub Access Policies
+
+<FeatureAvailability/>
+
+At the lowest level of DataHub's access control model is the concept of **Access Policies**. In conjunction with [Roles](./roles.md), Access Policies determine what users are allowed to do on DataHub.
+More specifically, Access Policies define *who* can *do what* to *which resources*.
```
[the remark this answers is not in the document]
Comment (maggiehays) [reply docs/authorization/access-policies-guide.md:6]: We should also define Platform & Metadata Policies in this section instead of below

File: docs/authorization/access-policies-guide.md [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+
+A few policies in plain English include
```
[the remark this answers is not in the document]
Comment (maggiehays) [reply docs/authorization/access-policies-guide.md:8]: If we define Platform & Metadata Policies in this section, rephase this to be: 

> Examples of Metadata Policies in plain English include:
>  Dataset Owners should be allowed to edit documentation, but not Tags. 
> ...

> Examples of Platform Policies in plain English include:
> The Data Platform team should be allowed to manage users & groups, view platform analytics, & manage policies themselves.

File: docs/authorization/access-policies-guide.md [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+
+* **Manage Policies** privilege
+
+This platform privilege allows users to create, edit, and remove all Access Policies on DataHub. Therefore, it should only be
+given to those users who will be serving as Admins of the platform. The default `Admin` role has this privilege.
+
+
+## Using Access Policies
+
+### Creating a new Policy
```
Comment (maggiehays) [inline docs/authorization/access-policies-guide.md:31]: Consider splitting this section into two -- `Creating a Metadata Policy` and `Creating a Platform Policy`; the current set of steps are hard to follow because there are nuances for each policy type & it's hard to know what is relevant to each use case

File: docs/authorization/access-policies-guide.md [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+Out of the box, DataHub is deployed with a set of pre-baked Policies. This set of policies serves the
+following purposes:
+
+1. Assigns immutable super-user privileges for the root `datahub` user account (Immutable)
+2. Assigns all Platform privileges for all Users by default (Editable)
+
+The reason for #1 is to prevent people from accidentally deleting all policies and getting locked out (`datahub` super user account can be a backup)
+The reason for #2 is to permit administrators to log in via OIDC or another means outside of the `datahub` root account
+when they are bootstrapping with DataHub. This way, those setting up DataHub can start managing policies without friction.
+Note that these privilege *can* and likely *should* be altered inside the **Policies** page.
```
Comment (maggiehays) [inline docs/authorization/access-policies-guide.md:278]: update to "this privilege"
> Note that these privilege *can* and likely *should* be altered inside the **Policies** page.

**Comment by maggiehays** [comment]
@jjoyce0510 Left comments throughout; this is a long read and is a bit confusing to jump between the nitty-gritty details between the policy types. Consider changing the outline to: 

# About DataHub Access Policies

Move description of & differences between Metadata Policy vs. Platform Policy here

## Access Policies Setup, Prerequisites, and Permissions

## Using Access Policies

### Creating a Metadata Policy

Step 1: Name & Describe
Step 2: Configure Privileges
Step 3: Configure Entity Types/Specific Assets
Step 4: Choose Policy Actors

### Creating a Platform Policy

Step 1: Name & Describe
Step 2: Configure Privileges
Step 3: Choose Policy Actors

### Updating an Existing Policy

### Removing a Policy

### Deactivating a Policy

### Default Policies

**Comment by laulpogan** [comment]
@shirshanka with both Maggie and John out we'll need to wait on this one, at least one of the open comments needs domain knowledge/attention

File: docs/authorization/access-policies-guide.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply docs/authorization/access-policies-guide.md:6]: i like this!

File: docs/authorization/access-policies-guide.md [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+A few policies in plain English include
+
+- Dataset Owners should be allowed to edit documentation, but not Tags.
+- Jenny, our Data Steward, should be allowed to edit Tags for any Dashboard, but no other metadata.
+- James, a Data Analyst, should be allowed to edit the Links for a specific Data Pipeline he is a downstream consumer of.
+- The Data Platform team should be allowed to manage users & groups, view platform analytics, & manage policies themselves.
+
+Each of these can be implemented by constructing DataHub Access Policies.
+
+In this guide, we'll describe how to create & manage Access Policies.
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply docs/authorization/access-policies-guide.md:17]: nope!

File: docs/authorization/access-policies-guide.md [hunk 10]
```diff
@@ -0,0 +1,312 @@
...
+
+## Access Policies Setup, Prerequisites, and Permissions
+
+What you need to manage Access Policies on DataHub:
+
+* **Manage Policies** privilege
+
+This platform privilege allows users to create, edit, and remove all Access Policies on DataHub. Therefore, it should only be
```
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply docs/authorization/access-policies-guide.md:25]: lets do upper case

File: docs/authorization/access-policies-guide.md (hunk 3 above)
Comment (jjoyce0510) [reply docs/authorization/access-policies-guide.md:133]: nice catch!
